pkg/kube/controllers/extendedstatefulset/reconciler.go

```diff
@@ -258,3 +282,77 @@ func (r *ReconcileExtendedStatefulSet) getActualStatefulSet(ctx context.Context,
 
 	return result, maxVersion, nil
 }
+
+// listStatefulSetVersions gets all StatefulSets' versions and ready status owned by the ExtendedStatefulSet
+func (r *ReconcileExtendedStatefulSet) listStatefulSetVersions(ctx context.Context, exStatefulSet *essv1a1.ExtendedStatefulSet) (map[int]bool, error) {
+	result := map[int]bool{}
+
+	statefulSets, err := r.listStatefulSets(ctx, exStatefulSet)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, statefulSet := range statefulSets {
+		strVersion, ok := statefulSet.Annotations[essv1a1.AnnotationVersion]
+		if !ok {
+			strVersion = "0"
+		}
+
+		version, err := strconv.Atoi(strVersion)
+		if err != nil {
+			return result, errors.Wrapf(err, "Version annotation is not an int: %s", strVersion)
+		}
+
+		ready, err := r.isStatefulSetReady(ctx, &statefulSet)
+		if err != nil {
+			return nil, err
+		}
+
+		if ready {
+			result[version] = true
+		} else {
+			result[version] = false
+		}
+	}
+
+	return result, nil
+}
+
+// isStatefulSetReady returns true if one owned Pod is running
+func (r *ReconcileExtendedStatefulSet) isStatefulSetReady(ctx context.Context, statefulSet *v1beta1.StatefulSet) (bool, error) {
+	podList := &v1.PodList{}
+	err := r.client.List(
+		ctx,
+		&client.ListOptions{
+			Namespace:     statefulSet.Namespace,
+			LabelSelector: labels.Everything(),
+		},
+		podList,
+	)
+	if err != nil {
+		return false, err
+	}
+
+	for _, pod := range podList.Items {
+		if metav1.IsControlledBy(&pod, statefulSet) {
+			if isPodReady(&pod) {
+				r.log.Debug("Pod '", statefulSet.Name, "' owned by StatefulSet '", statefulSet.Name, "' is running.")
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// isPodReady returns false if the Pod Status is nil
+func isPodReady(pod *v1.Pod) bool {
+	var condition *v1.PodCondition
+	for i := range pod.Status.Conditions {
```

Comment:
Could this maybe be simplified using [GetPodCondition](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/pod/util.go#L194)?